sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/NormalizePlan.scala

```diff
@@ -105,8 +104,15 @@ object NormalizePlan extends PredicateHelper {
       case Project(projectList, child) =>
         Project(normalizeProjectList(projectList), child)
       case c: KeepAnalyzedQuery => c.storeAnalyzedQuery()
-      case localRelation: LocalRelation =>
-        ComparableLocalRelation.fromLocalRelation(localRelation)
+      case localRelation: LocalRelation if !localRelation.data.isEmpty =>
+        /**
+         * A substitute for the [[LocalRelation.data]]. [[GenericInternalRow]] is incomparable for
+         * maps, because [[ArrayBasedMapData]] doesn't define [[equals]].
+         */
+        val unsafeProjection = UnsafeProjection.create(localRelation.schema)
```

Comment:
`LocalRelation#schema` is actually `DataTypeUtils.fromAttributes(output)`. I think `UnsafeProjection.create(localRelation.output, localRelation.output)` is more efficient.

Reply:
Sure, resolved.

Reply:
Actually, it doesn't work in the context of plan normalization. Plan normalization resets expression ids to 0, and `bindReferences` inside of this `create` overload always returns `BoundReference` with 0 ordinal.

I changed this code to  a previous version.